examples/nlp/semantic_similarity_with_bert.py

```diff
@@ -0,0 +1,363 @@
+        token_type_ids = np.array(batch_token_type_ids, dtype="int32")
+
+        # set to true if data generator is used for training/validation
+        if self.include_targets:
+            labels = np.array(self.labels[indexes], dtype="int32")
+            return [input_ids, masks, token_type_ids], labels
+        else:
+            return [input_ids, masks, token_type_ids]
+
+    def on_epoch_end(self):
+        # Shuffle indexes after each epoch if shuffle is set to True.
+        self.indexes = np.arange(len(self.sentence1))
+        if self.shuffle:
+            np.random.shuffle(self.indexes)
+
+
+"""
+## Build The Model
+"""
+
+
+def build_model():
+    # Encoded token ids from BERT tokenizer.
+    input_ids = tf.keras.layers.Input(
+        shape=(max_length,), dtype=tf.int32, name="input_ids"
+    )
+    # Attention masks indicates to the model which tokens should be attended to.
+    attention_masks = tf.keras.layers.Input(
+        shape=(max_length,), dtype=tf.int32, name="att_mask"
+    )
+    # Token type ids are binary masks identifying different sequences in the model.
+    token_type_ids = tf.keras.layers.Input(
+        shape=(max_length,), dtype=tf.int32, name="tt_ids"
+    )
+    # Loading pretrained BERT model.
+    bert_model = TFBertModel.from_pretrained("bert-base-uncased")
+    sequence_output, pooled_output = bert_model(
+        input_ids, attention_mask=attention_masks, token_type_ids=token_type_ids
+    )
+    # Applying hybrid pooling approach to bert sequence output.
+    avg_pool = tf.keras.layers.GlobalAveragePooling1D()(sequence_output)
+    max_pool = tf.keras.layers.GlobalMaxPooling1D()(sequence_output)
+    concat = tf.keras.layers.concatenate([avg_pool, max_pool])
+    dropout = tf.keras.layers.Dropout(0.1)(concat)
+    output = tf.keras.layers.Dense(3, activation="softmax")(dropout)
+
+    model = tf.keras.models.Model(
+        inputs=[input_ids, attention_masks, token_type_ids], outputs=output
+    )
+
+    model.compile(
+        optimizer=tf.keras.optimizers.Adam(learning_rate=learning_rate),
+        loss="categorical_crossentropy",
+        metrics=["acc"],
+    )
+    return model
+
+
+"""
+Recommended training on GPU Runtime
+"""
+# Build model with distributed strategy.
+strategy = tf.distribute.MirroredStrategy()
+with strategy.scope():
+    model = build_model()
+
+print(f"Strategy: {strategy}")
+model.summary()
+
+"""
+Create train and validation data generators
+"""
+train_data = BertSemanticDataGenerator(
+    train_df.sentence1.astype("str"),
+    train_df.sentence2.astype("str"),
+    y_train,
+    batch_size=batch_size,
+    shuffle=True,
+)
+valid_data = BertSemanticDataGenerator(
+    valid_df.sentence1.astype("str"),
+    valid_df.sentence2.astype("str"),
+    y_val,
+    batch_size=batch_size,
+    shuffle=False,
+)
+
+"""
+## Train the Model
+"""
+history = model.fit_generator(
+    train_data,
+    steps_per_epoch=len(train_data) // batch_size,
```

Comment:
Since a `Sequence` has a `__len__`, this argument should not be necessary (same for validation_steps)